### PR TITLE
Add long option names & update help message

### PR DESCRIPTION
**Pull request by sschr15**
This also may allow for easier work on combination flags, like perhaps a `--no-removal` which would apply `--no-remove-bridge --no-remove-synthetic --no-remove-getclass --no-remove-empty-try-catch --no-remove-imports` or something like that

### CODE REVIEW

File: src/org/jetbrains/java/decompiler/main/extern/IFernflowerPreferences.java [hunk 1]
```diff
@@ -14,207 +14,258 @@
 public interface IFernflowerPreferences {
   @Name("Remove Bridge Methods")
```
Comment (ByMartrixx) [inline src/org/jetbrains/java/decompiler/main/extern/IFernflowerPreferences.java:15]: The changes made here break any application that passes options to the BaseDecompiler constructor, like the Intellij plugin or Enigma

File: src/org/jetbrains/java/decompiler/main/decompiler/OptionParser.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.Map;
+
+public class OptionParser {
+  private static final Map<String, String> SHORT_TO_LONG_NAME_MAP = new HashMap<>();
+
+  private static final String[] CUSTOM_CHECKS = {
+    "-e=",
+    "--add-external=",
+    "-only=",
+    "--only=",
```
Comment (ByMartrixx) [inline src/org/jetbrains/java/decompiler/main/decompiler/OptionParser.java:16]: `--only=` isn't handled in `ConsoleDecompiler`

**Review by ByMartrixx — CHANGES_REQUESTED** [review]

**Review by ByMartrixx — APPROVED** [review]
Looks good!

**Review by Earthcomputer — APPROVED** [review]
These changes are also good for the intellij plugin. I didn't like having to guess types

**Review by jaskarth — APPROVED** [review]
LGTM, tysm for doing this! The README needs to be updated, but that can be lower priority.

**Pull request merged**